### PR TITLE
drivers/bluetooth: stm32 ipm: Fix mutex on shared memory

### PR DESCRIPTION
**Pull request by erwango**
ipm_busy is meant to be used as mutex on transport layer accesses.
Due to wrong configuration on rx_thread, imp_busy was a no-op.
Fix this.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

### CODE REVIEW

**Review by jhedberg — APPROVED** [review]

File: drivers/bluetooth/hci/ipm_stm32wb.c [hunk 1]
```diff
@@ -157,7 +157,7 @@ static void bt_ipm_rx_thread(void)
 
 		hcievt = k_fifo_get(&ipm_rx_events_fifo, K_FOREVER);
 
-		k_sem_take(&ipm_busy, K_NO_WAIT);
+		k_sem_take(&ipm_busy, K_FOREVER);
```
Comment (joerchan) [inline drivers/bluetooth/hci/ipm_stm32wb.c:160]: It looks like you have duplicate `k_sem_give` in the while loop here. I think maybe the ones in the switch statement can be removed.

Comment (erwango) [reply drivers/bluetooth/hci/ipm_stm32wb.c:160]: Thanks, actually it seems that the previous change broke the error management in this loop.
I've pushed a fix

File: drivers/bluetooth/hci/ipm_stm32wb.c [hunk 2]
```diff
@@ -168,8 +168,8 @@ static void bt_ipm_rx_thread(void)
 				/* Vendor events are currently unsupported */
 				BT_ERR("Unknown evtcode type 0x%02x",
 				       hcievt->evtserial.evt.evtcode);
-				k_sem_give(&ipm_busy);
-				break;
+				TL_MM_EvtDone(hcievt);
```
Comment (joerchan) [inline drivers/bluetooth/hci/ipm_stm32wb.c:171]: It looks like this wasn't here earlier, but it would make sense that it should be.

**Review by joerchan — APPROVED** [review]

File: drivers/bluetooth/hci/ipm_stm32wb.c (hunk 2 above)
Comment (erwango) [reply drivers/bluetooth/hci/ipm_stm32wb.c:171]: indeed, this was an issue in the initial version.

**Pull request merged**